spring-kafka/src/main/java/org/springframework/kafka/listener/AbstractMessageListenerContainer.java

```diff
@@ -197,10 +197,8 @@ public void setupMessageListener(Object messageListener) {
 	public final void start() {
 		synchronized (this.lifecycleMonitor) {
 			Assert.isTrue(
-					this.containerProperties.getMessageListener() instanceof MessageListener
-							|| this.containerProperties.getMessageListener() instanceof AcknowledgingMessageListener,
-					"Either a " + MessageListener.class.getName() + " or a "
-							+ AcknowledgingMessageListener.class.getName() + " must be provided");
+					this.containerProperties.getMessageListener() instanceof KafkaDataListener,
+					"A " + KafkaDataListener.class.getName() + " implementation must be provided");
```

Comment:
Not sure what is "STET", but eh, let it be. we can live without that coercion.
Let me come through the changes locally and I'll merge it if that.